commands/index.go

```diff
@@ -0,0 +1,202 @@
+package commands
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
+	lotuscli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-cid"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lily/chain/actors/builtin"
+	"github.com/filecoin-project/lily/chain/indexer"
+	"github.com/filecoin-project/lily/lens/lily"
+)
+
+type indexOps struct {
+	tasks    string
+	storage  string
+	apiAddr  string
+	apiToken string
+	name     string
+	window   time.Duration
+}
+
+var indexFlags indexOps
+
+var IndexTipSetCmd = &cli.Command{
+	Name:  "tipset",
+	Usage: "Index the state of a tipset from the filecoin blockchain by tipset key",
+	Action: func(cctx *cli.Context) error {
+		ctx := lotuscli.ReqContext(cctx)
+
+		indexName := fmt.Sprintf("index_%d", time.Now().Unix())
+		if indexFlags.name != "" {
+			indexName = indexFlags.name
+		}
+
+		var tsStr string
+		if tsStr = cctx.Args().First(); tsStr == "" {
+			return xerrors.Errorf("tipset argument required")
+		}
+
+		tsk, err := parseTipSetKey(strings.Split(tsStr, ","))
```

Comment:
Should the split be inside `parseTipSetKey`?

Reply:
Yea good idea